keras/layers/core/dense.py

```diff
@@ -201,6 +193,82 @@ def enable_lora(
         self.lora_enabled = True
         self.lora_rank = rank
 
+    def save_own_variables(self, store):
+        if not self.lora_enabled:
+            return super().save_own_variables(store)
+
+        kernel_value = self.kernel
+        store["0"] = kernel_value
+        if self.use_bias:
+            store["1"] = self.bias
+
+    def load_own_variables(self, store):
+        if not self.lora_enabled:
+            return super().load_own_variables(store)
+        self._kernel.assign(store["0"])
+        if self.use_bias:
+            self.bias.assign(store["1"])
+        self.lora_kernel_a.assign(np.zeros(self.lora_kernel_a.shape))
+        self.lora_kernel_b.assign(np.zeros(self.lora_kernel_b.shape))
+
+    def get_config(self):
+        base_config = super().get_config()
+        config = {
+            "units": self.units,
+            "activation": activations.serialize(self.activation),
+            "use_bias": self.use_bias,
+            "kernel_initializer": initializers.serialize(
+                self.kernel_initializer
+            ),
+            "bias_initializer": initializers.serialize(self.bias_initializer),
+            "kernel_regularizer": regularizers.serialize(
+                self.kernel_regularizer
+            ),
+            "bias_regularizer": regularizers.serialize(self.bias_regularizer),
+            "kernel_constraint": constraints.serialize(self.kernel_constraint),
+            "bias_constraint": constraints.serialize(self.bias_constraint),
+        }
+        if self.lora_rank:
+            config["lora_rank"] = self.lora_rank
+        return {**base_config, **config}
+
+    """Quantization-related methods"""
+
+    def quantized_build(self, input_shape, mode):
+        input_dim = input_shape[-1]
+        kernel_shape = (input_dim, self.units)
+        if mode == "int8":
+            self.inputs_quantizer = quantizers.AbsMaxQuantizer(axis=-1)
+            self._kernel = self.add_weight(
+                name="kernel",
+                shape=kernel_shape,
+                initializer="zeros",
+                dtype="int8",
+                trainable=False,
+            )
+            kernel_scale_shape = (1, kernel_shape[1])
+            self.kernel_scale = self.add_weight(
```

Comment:
> It should default to 1, not 0, otherwise the layer's output would always be 0.

I've changed the defaults from "zeros" to "ones". However, it should be considered a dummy initialization because the quantized weights must be loaded from a pretrained model.

> why does it need to be a variable? Could it just be a constant?

It would be reasonable to allow `kernel_scale` to be a variable. We can easily save and load it, and even modify it with the current APIs.

> And does it need to have shape (1, kernel_shape[1])? Could it be a scalar?
We are using per-channel dynamic quantization to minimize quantization errors.

In fact, the current implementation mirrors that of google/gemma_pytorch:
https://github.com/google/gemma_pytorch/blob/main/gemma/model.py#L112-L121